src/components_utils/express.jl

```diff
@@ -0,0 +1,105 @@
+import Base64
+export dcc_send_file, dcc_send_string, dcc_send_bytes
+"""
+    dbc_send_file(path::AbstractString, filename = nothing; type = nothing)
+
+Convert a file into the format expected by the Download component.
+
+# Arguments
+- `path` - path to the file to be sent
+- `filename` - name of the file, if not provided the original filename is used
+- `type` - type of the file (optional, passed to Blob in the javascript layer)
+"""
+function dcc_send_file(path, filename = nothing; type = nothing)
+    filename = isnothing(filename) ? basename(path) : filename
+    return dcc_send_bytes(read(path), filename, type = type)
+end
+
+"""
+    dcc_send_bytes(src::AbstractVector{UInt8}, filename; type = nothing)
+    dcc_send_bytes(src::Function, filename; type = nothing)
+
+Convert vector of bytes into the format expected by the Download component.
+
+# Examples
+
+Sending binary content
+```julia
+file_data = read("path/to/file")
+callback!(app, Output("download", "data"), Input("download-btn", "n_clicks"), prevent_initial_call = true) do n_clicks
+    return dcc_send_bytes(file_data, "filename.fl")
+end
+```
+
+Sending `DataFrame` in `Arrow` format
+```julia
+using DataFrames, Arrow
+...
+df = DataFrame(...)
+callback!(app, Output("download", "data"), Input("download-btn", "n_clicks"), prevent_initial_call = true) do n_clicks
+    return dcc_send_bytes("df.arr") do io
+        Arrow.write(io, df)
+    end
```

Comment:
This form where the first arg to `dcc_send_bytes` is a function - it's a bit more cumbersome and confusing than the Python version that [looks like](https://github.com/plotly/dash-core-components/pull/863)
```py
@app.callback(Output("download", "data"), Input("btn", "n_clicks"))
def func(n_clicks):
    return dcx.send_data_frame(df.to_csv, "mydf.csv")
```
The Python one is already a little unintuitive - I wish pandas would have just allowed `df.to_csv` to output a string if called with no buffer or file path - but at least you don't have to explicitly create a new function referencing an `io` that you don't care about. I think if I were using this I'd prefer to just write it out myself:

```julia
callback!(app, Output("download", "data"), Input("download-btn", "n_clicks"), prevent_initial_call = true) do n_clicks
    io = IOBuffer()
    Arrow.write(io, df)
    return dcc_send_bytes(take!(io), "df.arr")
```

More characters but the same number of lines in total, and less to learn about this interface. Is there any other way we can make this cleaner? Are there other use cases where the function to be called as `src(io)` is already defined, the way `df.to_csv` is? If not I might suggest dropping the `src::Function` form.

Reply:
`dcc_send_bytes(take!(io), "df.arr")` is also already possible. (`dcc_send_bytes(src::AbstractVector{UInt8}, filename; type = nothing)`)
The point of the version with `callback` and `io` is not to do work  for a large amount of data  like writing data to a binary buffer - turning it into an array - writing this array into another buffer - and again converting it into a string to send

Reply:
I did not make an analogue of `send_data_frame because` this would lead to a direct dependence of Dash on DataFrames and all packages of a particular serialization (CSV, Arrow, etc.) and thus would mean the need for a release every time the version of these packages changes

Reply:
at 925744e i change `dcc_send_bytes(src::Function, filename; type = nothing)` to `dcc_send_bytes(writer::Function, data, filename; type = nothing)` so example above is now look like
```
return dcc_send_bytes(Arrow.write, df, "df.arr")
```

Reply:
Much cleaner! 🎉